applications/serial_lte_modem/src/slm_at_tcp_proxy.c

```diff
@@ -53,19 +56,22 @@ enum slm_tcp_proxy_at_cmd_type {
 };
 
 /** forward declaration of cmd handlers **/
+static int handle_at_tcp_whitelist(enum at_cmd_type cmd_type);
 static int handle_at_tcp_server(enum at_cmd_type cmd_type);
 static int handle_at_tcp_client(enum at_cmd_type cmd_type);
 static int handle_at_tcp_send(enum at_cmd_type cmd_type);
 static int handle_at_tcp_recv(enum at_cmd_type cmd_type);
 
 /**@brief SLM AT Command list type. */
 static slm_at_cmd_list_t tcp_proxy_at_list[AT_TCP_PROXY_MAX] = {
+	{AT_TCP_WHITELIST, "AT#XTCPWHTLST", handle_at_tcp_whitelist},
 	{AT_TCP_SERVER, "AT#XTCPSVR", handle_at_tcp_server},
 	{AT_TCP_CLIENT, "AT#XTCPCLI", handle_at_tcp_client},
 	{AT_TCP_SEND, "AT#XTCPSEND", handle_at_tcp_send},
 	{AT_TCP_RECV, "AT#XTCPRECV", handle_at_tcp_recv},
 };
 
+static char ip_whitelist[CONFIG_SLM_WHITELIST_SIZE][INET_ADDRSTRLEN];
```

Comment:
It would be more efficient to store the IP addresses in its binary form (`struct in_addr`) - it would make this array four times smaller. The addresses provided by the command could be easily converted with `inet_pton`. It would also be faster to search the whitelist then.

Not sure how much RAM we have to spare, so I'll leave the decission whether to optimize up to you.

Reply:
Current code uses 6*15=90 bytes data area. By changing to binary form that's about 6*4=24bytes. As we have to call inet_ntop to get the string format in order to notify AT client, please allow us to keep the string-form comparison as it is.